core/base.py

```diff
@@ -0,0 +1,323 @@
+#########
+# Copyright (c) 2015 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  * See the License for the specific language governing permissions and
+#  * limitations under the License.
+
+from boto import exception
+from ec2 import utils as ec2_utils
+from ec2 import constants
+from vpc import connection
+from cloudify.exceptions import NonRecoverableError, RecoverableError
+from cloudify import ctx
+
+
+class AwsBase(object):
+
+    def __init__(self,
+                 client=None
+                 ):
+        self.client = client if client else connection.VPCConnectionClient().client()
+
+    def communicate(self, fn, args=None):
```

Comment:
consider renaming to 
def execute() <- we want to encapsulate internal comm
def send_to_amazon <- we want to hide external comm